### PR TITLE
Fix Circle

### PR DESCRIPTION
**Pull request by massich**

### CODE REVIEW

**Comment by massich** [comment]
Cool!!

**Comment by agramfort** [comment]
you need to update config.py to use mne sample data_path and rename sample-trans.fif to sample_audvis-trans.fif and it should work fin. (except the ICA + SSP scripts which are not done yet).

**Comment by massich** [comment]
@agramfort 	I didn't follow what you said. I added a 00-fetcher to make it work.

**Comment by agramfort** [comment]
Epochs evoked and report scripts should already work. Please add them to CI

File: .circleci/config.yml [hunk 1]
```diff
@@ -46,17 +46,6 @@ jobs:
...
-        # possibly because the libGL expects a sufficiently modern libgcc
-        # and conda rolls their own (libgcc-ng package).
-        - run:
-            name: Patch Qt
-            command: |
-              conda remove --yes qt pyqt matplotlib sip libxcb icu vtk
-              pip uninstall --yes mayavi vtk
-              pip install vtk mayavi PyQt5 PyQt5-sip sip matplotlib
-              echo "export LD_PRELOAD=~/miniconda/envs/mne/lib/libgobject-2.0.so.0" >> $BASH_ENV
-              pip install --upgrade "https://api.github.com/repos/nipy/PySurfer/zipball/master"
```
Comment (agramfort) [inline .circleci/config.yml]: You will need mayavi for the report script

Comment (massich) [reply .circleci/config.yml]: mayavi should be there already. But yes I need to understand why it was not passing.

File: 04-artifact_correction_ica.py [hunk 2]
```diff
@@ -12,44 +12,35 @@
...
-                                '_raw.fif' % (run, 1))
-        raws.append(mne.io.read_raw_fif(run_fname))
+    raws = [mne.io.read_raw_fif(raw_fnames)]
+
     raw = mne.concatenate_raws(raws)
     # SSS reduces the data rank and the noise levels, so let's include
     # components based on a higher proportion of variance explained (0.999)
     # than we would otherwise do for non-Maxwell-filtered raw data (0.98)
-    n_components = 0.999
+    n_components = 0.999  # XXX: This can bring troubles to ICA
```
Comment (massich) [inline 04-artifact_correction_ica.py]: The right way to set the n_components is this:
```py
meg_maxfilter_rank = raw.copy().pick_types(meg=True).estimate_rank()
ica = mne.preprocessing.ICA(n_components=meg_maxfilter_rank,
                            method=method,
                            random_state=42)
```

Plus there was the problem of centring the PCA. (check https://github.com/massich/epilepsy-biomag-2018)

File: 04-artifact_correction_ssp.py [hunk 3]
```diff
@@ -12,20 +12,28 @@
...
         # XXX TODO
         raw = mne.io.read_raw_fif(raw_fnames_in)
+        projs, events = compute_proj_ecg(raw, n_grad=1, n_mag=1, n_eeg=0, average=True)
+        ecg_projs = projs[-2:]
+        projs, events = compute_proj_eog(raw, n_grad=1, n_mag=1, n_eeg=1, average=True)
+        eog_projs = projs[-3:]
+
+        raw.info['projs'] += eog_projs + ecg_projs
+        raw.apply_proj()
+        raw.save(proj_fnames_out, overwrite=True)
```
Comment (massich) [inline 04-artifact_correction_ssp.py]: When I run this locally, I already have a file called `sample_audvis_filt-proj.fif` is it that mne already distributes a precomputed version of it? shouldn't the study write all the intermediate files somewhere out of the sample?

I'm confused now.

**Pull request merged**